src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -774,11 +775,14 @@ private void EnsureNonTypeMemberNamesAreLoaded()
 
                 try
                 {
-                    foreach (var methodDef in module.GetMethodsOfTypeOrThrow(_handle))
+                    foreach (var fieldDef in module.GetFieldsOfTypeOrThrow(_handle))
```

Comment:
the reordering makes this all more confusing :)  consider keeping the same order, and just tweaking the invidiaul cases. (haven't deeply reviewed this).

Reply:
We use hashset also because we want o(1) containment checks. 

Ordering of enumeration has never been something I think Roslyn guaranteed. I don't mind adding that. But it def feels optional (and not something I would work hard to support).